operator/controllers/csiscaleoperator_controller.go

```diff
@@ -446,18 +446,24 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 	if err == nil && len(cm.Data) != 0 {
-		cmData, daemonSetMaxUnavailable = parseConfigMap(cm)
+		cmData, daemonSetMaxUnavailable = r.parseConfigMap(instance, cm)
```

Comment:
We shouldn't validate the configmap data every time in the reconcile loop, validation should be done only when there is a change in the configmap data.